### PR TITLE
init commit

### PR DESCRIPTION
**Pull request by evan-forbes**
## Description

Initial commit that moves all of the code relevant to creating and signing transaction from the app to here. It also mirrors the consts package from celestia-core using a github actions bot. This is needed so that other repos, such as optiminit, can uses code to programmatically sign transactions without importing celestia-core.

After this PR is merged, we can one of two things. Either keep the transaction code in celestia-app and use a github bot to mirror changes from that repo to this one or we keep all of the code in here in a single location and delete the relevant code in the app.

### CODE REVIEW

File: .github/sync.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+celestiaorg/celestia-core:
+  - source: pkg/consts/consts.go
+    dest: payment/consts/celestia_core_consts.go
```
Comment (evan-forbes) [inline .github/sync.yml:3]: still need to try this with the CI, but this essentially attempts to copy over the celestia-core/pkg/consts package every day, if there are any differences, it opens a PR. this allows us to not import celestia-core, but still track the changes to constants

**Review by liamsi — COMMENTED** [review]
This looks good to me. I think we need a clear plan how to move forward with data availability sampling, optimint, and app?

**Comment by evan-forbes** [comment]
converted to draft as we will likely not need this repo anymore.

**Comment by evan-forbes** [comment]
While we originally created this repo for optimint, which we don't need anymore thanks to the dalc repo, we might actually end up needing this depending on the decision made in the [ABCI++ ADR](https://github.com/celestiaorg/celestia-app/blob/92341dd68ee6e555ec6c0bb780afa3a1c8243a93/adrs/adr008:adopt-ABC%2B%2B-early.md)

If we decide not to adopt ABCI++ before mainnet, then we will need to find a way to properly fill the data square and implement non-interactive defaults, one of the hacky alternatives involves parsing the transactions in celestia-core, and filling the block there.

**Comment by evan-forbes** [comment]
actually, nevermind, I don't think we need this anymore. If we can't wait for/implement ABCI++, then I don't think we should go down the super ultra hacky solution route anyway

**Pull request closed without merging**